### PR TITLE
ci: Build workspace.

### PR DESCRIPTION
**Pull request by duesee**
This changes the CI so that we build all members in the workspace with `--all-targets` (i.e., `--lib --bins --tests --benches --examples`) on ubuntu-latest. 

Previously we build `cli`, `ds-lib`, and `mls-ds` but, for example, not the `interop_client` (see https://github.com/openmls/openmls/pull/1350#pullrequestreview-1389406828) on ubuntu-latest and macos-latest.

### CODE REVIEW

File: .github/workflows/build_workspace.yml [hunk 1]
```diff
@@ -16,18 +16,11 @@ env:
...
 jobs:
   tests:
-    strategy:
-      fail-fast: false
-      matrix:
-        os:
-          - macos-latest
-          - ubuntu-latest
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
```
Comment (franziskuskiefer) [inline .github/workflows/build_workspace.yml:19]: Why did you remove macos? We should keep building on at least ubuntu and macos.

File: .github/workflows/ds.yml [hunk 2]
```diff
@@ -1,36 +0,0 @@
...
-          - windows-latest
-    runs-on: ${{ matrix.os }}
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - name: Test DS lib
-        run: cargo test --verbose -p ds-lib
```
Comment (franziskuskiefer) [inline .github/workflows/ds.yml:34]: We don't run these tests anymore. Not sure if that's a big deal for DS though.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, one nit and looks like the CI is failing.

File: .github/workflows/ds.yml (hunk 2 above)
Comment (duesee) [reply .github/workflows/ds.yml:34]: Good point. As discussed off-GitHub I'll add a `cargo test` afterwards.

**Review by franziskuskiefer — COMMENTED** [review]
Thanks, what are the crash files in `fuzz`?

**Comment by duesee** [comment]
> Thanks, what are the crash files in fuzz?

Forgot to add `--exclude` and libFuzzer doesn't like to be called through `cargo test`. Removed the files.

**Review by franziskuskiefer — APPROVED** [review]
Thanks!
